javascript/ql/src/semmle/javascript/security/dataflow/TaintedPath.qll

```diff
@@ -93,13 +93,38 @@ module TaintedPath {
         |
           name = argumentlessMethodName
         )
-        or
+      )
+      or
+      // array method calls of interest
+      exists(DataFlow::MethodCallNode mcn, string name | dst = mcn and mcn.calls(src, name) |
+        // A `str.split()` call can either split into path elements (`str.split("/")`) or split by some other string.
         name = "split" and
-        not exists(DataFlow::Node splitBy | splitBy = mcn.getArgument(0) |
-          splitBy.mayHaveStringValue("/") or
-          any(DataFlow::RegExpLiteralNode reg | reg.getRoot().getAMatchedString() = "/")
-              .flowsTo(splitBy)
+        (
+          if
+            exists(DataFlow::Node splitBy | splitBy = mcn.getArgument(0) |
+              splitBy.mayHaveStringValue("/") or
+              any(DataFlow::RegExpLiteralNode reg | reg.getRoot().getAMatchedString() = "/")
+                  .flowsTo(splitBy)
+            )
+          then
+            srclabel.(Label::PosixPath).canContainDotDotSlash() and
+            dstlabel instanceof Label::SplitPath
+          else srclabel = dstlabel
         )
+        or
+        (
+          name = "pop" or
```

Comment:
`pop` is a bit surprising here since the result isn't an array and is thus unlikely to flow to a `join` call where it can "escape" the splitPath flow label.

It would be interesting to try the flow label mapping `splitPath -> normalizedRelativePath` instead, although maybe not worth running another evaluation for that.